### PR TITLE
Feature/#55 avoid unwrap expect

### PR DESCRIPTION
**Pull request by kitzbergerg**
Deny expect and unwrap like discussed in #55. For now allow already existing expect. Depending on whether those can actually fail we might later on remove those as well.

Fix other clippy warnings as well as they would be quite annoying when adding a clippy check to ci.

### CODE REVIEW

File: backend/src/services/seed_service.rs [hunk 1]
```diff
@@ -9,6 +9,7 @@ use crate::{
     },
 };
 
+#[allow(clippy::expect_used)]
 pub fn find_all(pool: &Data<Pool>) -> Result<Vec<SeedDTO>, ServiceError> {
     let mut conn = pool.get().expect("Failed to retrieve pool.");
```
Comment (markus2330) [inline backend/src/services/seed_service.rs:14]: Is there maybe some way to reconnect?

**Review by markus2330 — CHANGES_REQUESTED** [review]
Thank you for the PR! :rocket:

> Deny expect and unwrap like discussed in https://github.com/ElektraInitiative/PermaplanT/issues/55. For now allow already existing expect. Depending on whether those can actually fail we might later on remove those as well.

Please actually check (by docu etc.) if they can fail. Every `allow(clippy` should have a comment why and for which specific purpose there is this exception.

See #60.

> Fix other clippy warnings as well as they would be quite annoying when adding a clippy check to ci.

Are there still clippy warnings? If so, please create an issue (and let us keep this PR short).

**Comment by kitzbergerg** [comment]
> Please actually check (by docu etc.) if they can fail. Every allow(clippy should have a comment why and for which specific purpose there is this exception.

There is now only one exception where expect is used and that is when setting up the initial connection.

> Are there still clippy warnings? If so, please create an issue (and let us keep this PR short).

I fixed them as part of this PR (there we only like 3 errors and all of just something like using return ...; instead of implicit return

**Comment by markus2330** [comment]
Thank you, great job!

**Pull request merged**